hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemCreate.java

```diff
@@ -329,7 +330,7 @@ public void testMkdirOnRoot() throws Exception {
     final Path path = new Path("/");
     fs.setWorkingDirectory(new Path("/"));
     fs.mkdirs(path);
-    // Todo: Add assert after fixing getBlobProperties on root @anujmodi2011
+    Assert.assertTrue(fs.getAbfsStore().getBlobProperty(path, Mockito.mock(TracingContext.class)).getIsDirectory());
```

Comment:
lets check if this will work on dfs endpoint and for hns account.

if not, we would have to something like if-else. If yes, lets not make change to `getBlobProperty`.

Reply:
This is not working on DFS. For DFS we have to use getFileStatus().
Added if else here for that.

As we discussed offline, we may skip changing getBlobProperty because it is written such that it will never be called on root. Only place where we are calling it on root was a testcase.

Let's discuss this with other's as well.

Reply:
In case it is just called from a test case, it is better in my opinion to not change the method itself. We can check how we can expose getContainerProperty call for test?

Reply:
@anmolanmol1234 @snvijaya 
Your thoughts on this??

Reply:
Directly calling getContainerProperty for root on Blob endpoint